File: Algorithm/Baekjoon/Silver/[1262번] 알파벳 다이아몬드.py
import sys

input = sys.stdin.readline

N, R1, C1, R2, C2 = map(int, input().split())

alphabet = [chr(ord("a")+i) for i in range(N)]
dia = 2*N-1
for i in range(R1, R2+1):
    for j in range(C1, C2+1):
        i %= dia
        j %= dia
        dis = abs(N-1-i) + abs(N-1-j)
        if dis >  N-1:
            print(".", end="")
        else:
            print(alphabet[dis%26], end="")
    print()

File: Algorithm/Baekjoon/Silver/[1262번] 알파벳 다이아몬드.java
import java.io.*;
import java.util.*;

//import 해줘야함
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(st.nextToken());
        int R1 = Integer.parseInt(st.nextToken());
        int C1 = Integer.parseInt(st.nextToken());
        int R2 = Integer.parseInt(st.nextToken());
        int C2 = Integer.parseInt(st.nextToken());
        ArrayList<Character> alphabet = new ArrayList<>();
        int a = 'a';
        for (int i = 0; i < N; i++) {
            alphabet.add((char) (a + i));
        }
        int dia = 2 * N - 1;
        for (int i = R1; i <= R2; i++) {
            for (int j = C1; j <= C2; j++) {
                int x = i % dia;
                int y = j % dia;
                int dis = Math.abs(N - 1 - x) + Math.abs(N - 1 - y);
                sb.append((dis > N - 1) ? '.' : alphabet.get(dis % 26));
            }
            sb.append('\n');
        }
        System.out.println(sb);
    }
}